link/platform.go

```diff
@@ -0,0 +1,68 @@
+package link
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"runtime"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+const (
+	// Since 4.17, syscalls symbols are generated with the `__x64_` prefix.
+	// https://github.com/torvalds/linux/commit/d5a00528b58cdb2c71206e18bd021e34c4eab878
+	LinuxVersionWithPrefix = 4<<16 + 17<<8 + 0
```

Comment:
Version comparisons are brittle, since individual distributions freeze or change version numbers according to their own whims. It's always preferable to attempt an action and fall back to old behaviour if necessary.